src/js/configuration_panel/button-listeners.js

```diff
@@ -141,25 +201,39 @@ https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
         });
 
         // Remove the old configuration from Chrome's localStorage.
-        chrome.storage.local.remove(["gamepadConfiguration"], function () {
-            // Write the new gamepadConfiguration in localStorage.
-            chrome.storage.local.set({ gamepadConfiguration: gamepadConfiguration }, that.toggleSaveButtonState);
+        chrome.storage.local.remove([configurationName], function () {
+            var configurationWrapper = {};
+            configurationWrapper[configurationName] = gamepadConfiguration;
+
+            // Save the new configuration.
+            chrome.storage.local.set(configurationWrapper, function () {
+                /**
+                 * Toggle (disable) the buttons if the data is stored as
+                 * "gamepadConfiguration" and not the unsaved changes.
+                 */
+                if (configurationName === "gamepadConfiguration") {
+                    that.toggleSaveAndDiscardButtons();
+                }
+            });
         });
     };
 
     /**
      *
-     * Toggle the "Save Changes" button when the input configuration is changed.
+     * Toggle the "Save Changes" and "Discard Changes" buttons when the input
+     * configuration is changed.
      *
      * @param {Object} that - The configurationPanel component.
      * @param {Object} saveChangesButton - The "Save Changes" button on the panel.
+     * @param {Object} discardButton - The "Discard Changes" button on the panel.
      *
      */
-    gamepad.configurationPanel.buttonListeners.toggleSaveButton = function (that, saveChangesButton) {
+    gamepad.configurationPanel.buttonListeners.toggleSaveAndDiscardButtons = function (that, saveChangesButton, discardButton) {
+        saveChangesButton = saveChangesButton[0];
+        discardButton = discardButton[0];
         chrome.storage.local.get(["gamepadConfiguration"], function (gamepadConfigurationWrapper) {
             // Get the list of all dropdowns in the configuration panel.
             var configurationMenus = document.querySelectorAll(".menu-item");
```

Comment:
So, this is what I meant about needing `fluid-binder`, right now the DOM elements are the only place the state is stored until it gets saved.   It'd be better to have form changes relayed to a "working copy" model, IMO.

Reply:
Please add a TODO somewhere nearby referencing this ticket:

https://github.com/fluid-lab/gamepad-navigator/issues/41